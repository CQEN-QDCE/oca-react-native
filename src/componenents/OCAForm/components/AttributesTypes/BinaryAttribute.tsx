import React from 'react';
import {
  Image,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
} from 'react-native';
import type { Attribute } from '../../../types';
import { attributeTypeStyle } from '../../../utils/attributeTypeStyle';

interface BinaryAttributeProps {
  attribute: Attribute;
  shown?: boolean;
  binaryStyle?: StyleProp<any>;
  textStyle?: StyleProp<TextStyle>;
}

export const BinaryAttribute = ({
  attribute,
  shown = false,
  binaryStyle,
  textStyle,
}: BinaryAttributeProps) => {
  const defaultStyle = StyleSheet.create({
    image: {
      height: 150,
      aspectRatio: 1,
      resizeMode: 'contain',
      borderRadius: 10,
      ...binaryStyle,
    },
  });

  return (
    <View style={attributeTypeStyle.textValueContainer}>
      {shown ? (
        <Image style={defaultStyle.image} source={{ uri: attribute.value }} />
      ) : (
        <Text style={[attributeTypeStyle.text, textStyle]}>
          {Array(10).fill('\u2022').join('')}
        </Text>
      )}
    </View>
  );
};
