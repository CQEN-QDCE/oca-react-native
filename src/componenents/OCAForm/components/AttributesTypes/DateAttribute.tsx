import React, { useEffect, useState } from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../types';
import { attributeTypeStyle } from '../../utils/attributeTypeStyle';
import moment from 'moment';

interface DateAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const DateAttribute = ({
  attribute,
  shown = false,
  styles,
}: DateAttributeProps) => {
  const [date, setDate] = useState<string>();

  useEffect(() => {
    let dataValue: Date | moment.Moment = new Date(attribute.value);
    if (!dataValue && moment(attribute.value, attribute.format).isValid()) {
      dataValue = moment(attribute.value, attribute.format);
    } else {
      return setDate(attribute.value);
    }
    setDate(moment(dataValue).format(attribute.format));
  }, [attribute.value, attribute.format]);

  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]}>
        {shown ? date : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
