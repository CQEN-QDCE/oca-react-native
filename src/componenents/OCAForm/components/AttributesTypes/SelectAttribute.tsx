import React from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../types';
import { attributeTypeStyle } from '../../utils/attributeTypeStyle';

interface SelectAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const SelectAttribute = ({
  attribute,
  shown = false,
  styles,
}: SelectAttributeProps) => {
  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]}>
        {shown ? attribute.value : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
