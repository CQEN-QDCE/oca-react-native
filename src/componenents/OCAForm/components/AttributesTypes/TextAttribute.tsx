import React from 'react';
import { StyleProp, Text, TextStyle, View } from 'react-native';
import type { Attribute } from '../../../types';
import { attributeTypeStyle } from '../../../utils/attributeTypeStyle';

interface TextAttributeProps {
  attribute: Attribute;
  shown: boolean;
  styles: StyleProp<TextStyle>;
}

export const TextAttribute = ({
  attribute,
  shown = false,
  styles,
}: TextAttributeProps) => {
  return (
    <View style={attributeTypeStyle.textValueContainer}>
      <Text style={[attributeTypeStyle.text, styles]} numberOfLines={200}>
        {shown ? attribute.value : Array(10).fill('\u2022').join('')}
      </Text>
    </View>
  );
};
