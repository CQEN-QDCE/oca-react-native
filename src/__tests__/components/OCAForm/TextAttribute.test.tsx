import renderer from 'react-test-renderer';
import React from 'react';
import { StyleSheet } from 'react-native';
import { TextAttribute } from '../../../components/OCAForm/components/AttributesTypes/TextAttribute';

const attribute = { name: 'Birth date', value: '2000-02-21', type: 'Text' };

describe('TextAttribute', () => {
  it('Given an attribute should render the attribute value properly.', () => {
    const tree = renderer
      .create(
        <TextAttribute attribute={attribute} shown={true} styles={undefined} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Given an attribute with custom styling should render the label with custom styling properly.', () => {
    const styles = StyleSheet.create({
      value: {
        fontSize: 32,
        color: 'red',
      },
    });
    const tree = renderer
      .create(
        <TextAttribute
          attribute={attribute}
          shown={true}
          styles={styles.value}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
