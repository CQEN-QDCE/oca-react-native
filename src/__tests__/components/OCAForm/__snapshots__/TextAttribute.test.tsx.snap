// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAttribute Given an attribute should render the attribute value properly. 1`] = `
<View
  style={
    Object {
      "color": "#333",
      "paddingVertical": 4,
    }
  }
>
  <Text
    numberOfLines={200}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    2000-02-21
  </Text>
</View>
`;

exports[`TextAttribute Given an attribute with custom styling should render the label with custom styling properly. 1`] = `
<View
  style={
    Object {
      "color": "#333",
      "paddingVertical": 4,
    }
  }
>
  <Text
    numberOfLines={200}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "color": "red",
          "fontSize": 32,
        },
      ]
    }
  >
    2000-02-21
  </Text>
</View>
`;
